src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -455,6 +430,44 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
             return ReadFunctions(metadatas, descriptorProviders);
         }
 
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+        {
+            if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
+            }
+            else
+            {
+                // First see if there is an explicit primary file indicated
+                // in config. If so use that.
+                string functionPrimary = null;
+                string scriptFileName = (string)functionConfig["scriptFile"];
+                if (!string.IsNullOrEmpty(scriptFileName))
+                {
+                    // allow the config file to contain just the file name, or file name
+                    // with extension
+                    scriptFileName = Path.GetFileNameWithoutExtension(scriptFileName);
+
+                    functionPrimary = functionFiles.FirstOrDefault(p =>
+                        string.Compare(Path.GetFileNameWithoutExtension(p), scriptFileName, StringComparison.OrdinalIgnoreCase) == 0);
```

Comment:
Well, to avoid all of this, I can just make source file have to be the entire file name. Perhaps that is cleaner and easier to explain.